File: Recursion/count_zeros.cpp
// Count the number of 0's in an integer
// Sample Input : 10010
// Output: 3
#include<bits/stdc++.h>
using namespace std;
int count_zeros(int n){
    if(n == 0)
        return 0;
    // divide the number recursively and hit the base case    
    int small_no = count_zeros(n / 10);
    int last_digit = n % 10;
    if(last_digit == 0)
        return small_no + 1;
    else 
        return small_no;    
}
int main(){
    cout << count_zeros(2003010);
    return 0;
}

File: Recursion/count_zeros.go
// Count the number of 0's in an integer
// Sample Input : 10010
// Output: 3
package main

import "fmt"

func CountZeroes(n int) int {
	if n == 0 {
		return 0;
	}
	// divide the number recursively and hit the base case
	smallNo := CountZeroes(n / 10)
	lastDigit := n % 10
	if lastDigit == 0 {
		smallNo += 1
	} else {
		return smallNo
	}
	return smallNo
}

func main() {
	msg := CountZeroes(1000100100100)
	fmt.Println(msg)
}